--- ./grbl/grbl.h	2017-07-09 10:15:23.439678589 -0700
+++ ./grbl/grbl.h	2017-07-09 10:17:11.712435811 -0700
@@ -26,11 +26,6 @@
 #define GRBL_VERSION_BUILD "20170131"
 
 // Define standard libraries used by Grbl.
-#include <avr/io.h>
-#include <avr/pgmspace.h>
-#include <avr/interrupt.h>
-#include <avr/wdt.h>
-#include <util/delay.h>
 #include <math.h>
 #include <inttypes.h>
 #include <string.h>
@@ -60,6 +55,7 @@
 #include "spindle_control.h"
 #include "stepper.h"
 #include "jog.h"
+#include "delay.h"
 
 // ---------------------------------------------------------------------------------------
 // COMPILE-TIME ERROR CHECKING OF DEFINE VALUES:
