Remove AVR specific include files.
--- grbl-master/grbl.h	2015-07-30 07:26:08.451929661 -0700
+++ grbl-master/grbl.h	2015-07-30 07:26:24.184289675 -0700
@@ -26,11 +26,11 @@
 #define GRBL_VERSION_BUILD "20150717"
 
 // Define standard libraries used by Grbl.
-#include <avr/io.h>
-#include <avr/pgmspace.h>
-#include <avr/interrupt.h>
-#include <avr/wdt.h>
-#include <util/delay.h>
+//#include <avr/io.h>
+//#include <avr/pgmspace.h>
+//#include <avr/interrupt.h>
+//#include <avr/wdt.h>
+//#include <util/delay.h>
 #include <math.h>
 #include <inttypes.h>    
 #include <string.h>
