Turn the GPIO control into board specific inline functions.
--- grbl-master/coolant_control.c	2014-04-01 20:41:19.000000000 -0700
+++ grbl-master/coolant_control.c	2014-04-01 20:43:21.000000000 -0700
@@ -30,19 +30,15 @@
 void coolant_init()
 {
   current_coolant_mode = COOLANT_DISABLE;
-  #if ENABLE_M7
-    COOLANT_MIST_DDR |= (1 << COOLANT_MIST_BIT);
-  #endif
-  COOLANT_FLOOD_DDR |= (1 << COOLANT_FLOOD_BIT);
   coolant_stop();
 }
 
 void coolant_stop()
 {
   #ifdef ENABLE_M7
-    COOLANT_MIST_PORT &= ~(1 << COOLANT_MIST_BIT);
+  coolant_mist_off();
   #endif
-  COOLANT_FLOOD_PORT &= ~(1 << COOLANT_FLOOD_BIT);
+  coolant_flood_off();
 }
 
 
@@ -52,10 +48,10 @@
   { 
     plan_synchronize(); // Ensure coolant turns on when specified in program.
     if (mode == COOLANT_FLOOD_ENABLE) { 
-      COOLANT_FLOOD_PORT |= (1 << COOLANT_FLOOD_BIT);
+      coolant_flood_on();
     #ifdef ENABLE_M7  
       } else if (mode == COOLANT_MIST_ENABLE) {
-          COOLANT_MIST_PORT |= (1 << COOLANT_MIST_BIT);
+          coolant_mist_on();
     #endif
     } else {
       coolant_stop();
