--- ./grbl/coolant_control.c	2017-07-09 10:36:48.136663290 -0700
+++ ./grbl/coolant_control.c	2017-07-09 10:43:10.163335040 -0700
@@ -23,10 +23,6 @@
 
 void coolant_init()
 {
-  COOLANT_FLOOD_DDR |= (1 << COOLANT_FLOOD_BIT); // Configure as output pin
-  #ifdef ENABLE_M7
-    COOLANT_MIST_DDR |= (1 << COOLANT_MIST_BIT);
-  #endif
   coolant_stop();
 }
 
@@ -35,19 +31,12 @@
 uint8_t coolant_get_state()
 {
   uint8_t cl_state = COOLANT_STATE_DISABLE;
-  #ifdef INVERT_COOLANT_FLOOD_PIN
-    if (bit_isfalse(COOLANT_FLOOD_PORT,(1 << COOLANT_FLOOD_BIT))) {
-  #else
-    if (bit_istrue(COOLANT_FLOOD_PORT,(1 << COOLANT_FLOOD_BIT))) {
-  #endif
+
+  if (coolant_flood_is_on()) {
     cl_state |= COOLANT_STATE_FLOOD;
   }
   #ifdef ENABLE_M7
-    #ifdef INVERT_COOLANT_MIST_PIN
-      if (bit_isfalse(COOLANT_MIST_PORT,(1 << COOLANT_MIST_BIT))) {
-    #else
-      if (bit_istrue(COOLANT_MIST_PORT,(1 << COOLANT_MIST_BIT))) {
-    #endif
+    if (coolant_mist_is_on()) {
       cl_state |= COOLANT_STATE_MIST;
     }
   #endif
@@ -59,18 +48,8 @@
 // an interrupt-level. No report flag set, but only called by routines that don't need it.
 void coolant_stop()
 {
-  #ifdef INVERT_COOLANT_FLOOD_PIN
-    COOLANT_FLOOD_PORT |= (1 << COOLANT_FLOOD_BIT);
-  #else
-    COOLANT_FLOOD_PORT &= ~(1 << COOLANT_FLOOD_BIT);
-  #endif
-  #ifdef ENABLE_M7
-    #ifdef INVERT_COOLANT_MIST_PIN
-      COOLANT_MIST_PORT |= (1 << COOLANT_MIST_BIT);
-    #else
-      COOLANT_MIST_PORT &= ~(1 << COOLANT_MIST_BIT);
-    #endif
-  #endif
+  coolant_flood_off();
+  coolant_mist_off();
 }
 
 
@@ -89,20 +68,12 @@
   } else {
   
     if (mode & COOLANT_FLOOD_ENABLE) {
-      #ifdef INVERT_COOLANT_FLOOD_PIN
-        COOLANT_FLOOD_PORT &= ~(1 << COOLANT_FLOOD_BIT);
-      #else
-        COOLANT_FLOOD_PORT |= (1 << COOLANT_FLOOD_BIT);
-      #endif
+      coolant_flood_on();
     }
   
     #ifdef ENABLE_M7
       if (mode & COOLANT_MIST_ENABLE) {
-        #ifdef INVERT_COOLANT_MIST_PIN
-          COOLANT_MIST_PORT &= ~(1 << COOLANT_MIST_BIT);
-        #else
-          COOLANT_MIST_PORT |= (1 << COOLANT_MIST_BIT);
-        #endif
+        coolant_mist_on();
       }
     #endif

