--- grbl-master/motion_control.c	2014-04-08 22:52:02.000000000 -0700
+++ grbl-master/motion_control.c	2014-04-08 22:53:42.000000000 -0700
@@ -33,6 +33,7 @@
 #include "planner.h"
 #include "limits.h"
 #include "protocol.h"
+#include "delay.h"
 
 // Execute linear motion in absolute millimeter coordinates. Feed rate given in millimeters/second
 // unless invert_feed_rate is true. Then the feed_rate means that the motion should be completed in
@@ -213,7 +214,7 @@
 void mc_go_home()
 {
   sys.state = STATE_HOMING; // Set system state variable
-  LIMIT_PCMSK &= ~LIMIT_MASK; // Disable hard limits pin change register for cycle duration
+  limits_enabled = 0; // Disable hard limits pin change register for cycle duration
   
   limits_go_home(); // Perform homing routine.
 
@@ -252,7 +253,7 @@
   sys_sync_current_position();
 
   // If hard limits feature enabled, re-enable hard limits pin change register after homing cycle.
-  if (bit_istrue(settings.flags,BITFLAG_HARD_LIMIT_ENABLE)) { LIMIT_PCMSK |= LIMIT_MASK; }
+  if (bit_istrue(settings.flags,BITFLAG_HARD_LIMIT_ENABLE)) { limits_enabled = 1; }
   // Finished! 
 }
 
