--- grbl-master/motion_control.c	2015-01-25 07:43:00.508659352 -0800
+++ grbl-master/motion_control.c	2015-01-25 08:58:10.837118132 -0800
@@ -33,6 +33,7 @@
 #include "planner.h"
 #include "limits.h"
 #include "protocol.h"
+#include "delay.h"
 
 // Execute linear motion in absolute millimeter coordinates. Feed rate given in millimeters/second
 // unless invert_feed_rate is true. Then the feed_rate means that the motion should be completed in
@@ -213,7 +214,7 @@
 void mc_go_home()
 {
   sys.state = STATE_HOMING; // Set system state variable
-  LIMIT_PCMSK &= ~LIMIT_MASK; // Disable hard limits pin change register for cycle duration
+  limits_enabled = 0; // Disable hard limits pin change register for cycle duration
   
   limits_go_home(); // Perform homing routine.
 
@@ -222,7 +223,7 @@
 
   // The machine should now be homed and machine zero has been located. Upon completion, 
   // reset system position and sync internal position vectors.
-  clear_vector_float(sys.position); // Set machine zero
+  clear_vector(sys.position); // Set machine zero
   sys_sync_current_position();
   sys.state = STATE_IDLE; // Set system state to IDLE to complete motion and indicate homed.
   
@@ -252,7 +253,7 @@
   sys_sync_current_position();
 
   // If hard limits feature enabled, re-enable hard limits pin change register after homing cycle.
-  if (bit_istrue(settings.flags,BITFLAG_HARD_LIMIT_ENABLE)) { LIMIT_PCMSK |= LIMIT_MASK; }
+  if (bit_istrue(settings.flags,BITFLAG_HARD_LIMIT_ENABLE)) { limits_enabled = 1; }
   // Finished! 
 }
 
