TODO
--- grbl-master/protocol.c	2014-03-16 22:11:20.000000000 -0700
+++ grbl-master/protocol.c	2014-03-16 22:12:28.000000000 -0700
@@ -48,11 +48,13 @@
 {
   protocol_reset_line_buffer();
   report_init_message(); // Welcome message   
-  
+
+#if 0  
   PINOUT_DDR &= ~(PINOUT_MASK); // Set as input pins
   PINOUT_PORT |= PINOUT_MASK; // Enable internal pull-up resistors. Normal high operation.
   PINOUT_PCMSK |= PINOUT_MASK;   // Enable specific pins of the Pin Change Interrupt
   PCICR |= (1 << PINOUT_INT);   // Enable Pin Change Interrupt
+#endif
 }
 
 // Executes user startup script, if stored.
@@ -71,6 +73,7 @@
   }  
 }
 
+#if 0
 // Pin change interrupt for pin-out commands, i.e. cycle start, feed hold, and reset. Sets
 // only the runtime command execute variable to have the main program execute these when 
 // its ready. This works exactly like the character-based runtime commands when picked off
@@ -88,6 +91,7 @@
     }
   }
 }
+#endif
 
 // Executes run-time commands, when required. This is called from various check points in the main
 // program, primarily where there may be a while loop waiting for a buffer to clear space or any
