--- ./grbl/main.c	2017-07-09 11:08:40.578038201 -0700
+++ ./grbl/main.c	2017-07-09 11:09:27.274364778 -0700
@@ -33,7 +33,7 @@
 volatile uint8_t sys_rt_exec_accessory_override; // Global realtime executor bitflag variable for spindle/coolant overrides.
 
 
-int main(void)
+int grbl_main(void)
 {
   // Initialize system upon power-up.
   serial_init();   // Setup serial baud rate and interrupts
@@ -42,7 +42,6 @@
   system_init();   // Configure pinout pins and pin-change interrupt
 
   memset(sys_position,0,sizeof(sys_position)); // Clear machine position.
-  sei(); // Enable interrupts
 
   // Initialize system state.
   #ifdef FORCE_INITIALIZATION_ALARM
