Turn the GPIO control into board specific inline functions.
--- grbl_master/spindle_control.c	2014-04-01 20:57:10.000000000 -0700
+++ grbl-master/spindle_control.c	2014-04-01 20:59:29.000000000 -0700
@@ -28,14 +28,12 @@
 void spindle_init()
 {
   current_direction = 0;
-  SPINDLE_ENABLE_DDR |= (1<<SPINDLE_ENABLE_BIT);
-  SPINDLE_DIRECTION_DDR |= (1<<SPINDLE_DIRECTION_BIT);  
   spindle_stop();
 }
 
 void spindle_stop()
 {
-  SPINDLE_ENABLE_PORT &= ~(1<<SPINDLE_ENABLE_BIT);
+  spindle_off();
 }
 
 void spindle_run(int8_t direction) //, uint16_t rpm) 
@@ -44,11 +42,11 @@
     plan_synchronize();
     if (direction) {
       if(direction > 0) {
-        SPINDLE_DIRECTION_PORT &= ~(1<<SPINDLE_DIRECTION_BIT);
+        spindle_fwd();
       } else {
-        SPINDLE_DIRECTION_PORT |= 1<<SPINDLE_DIRECTION_BIT;
+        spindle_rev();
       }
-      SPINDLE_ENABLE_PORT |= 1<<SPINDLE_ENABLE_BIT;
+      spindle_on();
     } else {
       spindle_stop();     
     }
