--- ./grbl/print.h	2017-07-09 11:41:26.339786014 -0700
+++ ./grbl/print.h	2017-07-09 11:42:25.184197553 -0700
@@ -34,6 +34,8 @@
 // Prints an uint8 variable in base 10.
 void print_uint8_base10(uint8_t n);
 
+void print_uint16_base10(uint16_t n);
+
 // Prints an uint8 variable in base 2 with desired number of desired digits.
 void print_uint8_base2_ndigit(uint8_t n, uint8_t digits);
 
--- ./grbl/print.c	2017-01-31 17:46:20.000000000 -0800
+++ ./grbl/print.c	2017-07-10 08:04:00.317041243 -0700
@@ -32,9 +32,8 @@
 // Print a string stored in PGM-memory
 void printPgmString(const char *s)
 {
-  char c;
-  while ((c = pgm_read_byte_near(s++)))
-    serial_write(c);
+  while (*s)
+    serial_write(*s++);
 }
 
 
@@ -113,6 +112,10 @@
     serial_write('0' + buf[i-1]);
 }
 
+void print_uint16_base10(uint16_t n)
+{
+  print_uint32_base10(n);
+}
 
 void printInteger(long n)
 {
