--- grbl-master/settings.h	2015-07-17 07:19:57.000000000 -0700
+++ grbl-master/settings.h	2015-07-31 21:42:33.312969516 -0700
@@ -85,8 +85,8 @@
 
   // Remaining Grbl settings
   uint8_t pulse_microseconds;
-  uint8_t step_invert_mask;
-  uint8_t dir_invert_mask;
+  uint16_t step_invert_mask;
+  uint16_t dir_invert_mask;
   uint8_t stepper_idle_lock_time; // If max value 255, steppers do not disable.
   uint8_t status_report_mask; // Mask to indicate desired report data.
   float junction_deviation;
@@ -94,7 +94,7 @@
   
   uint8_t flags;  // Contains default boolean settings
 
-  uint8_t homing_dir_mask;
+  uint16_t homing_dir_mask;
   float homing_feed_rate;
   float homing_seek_rate;
   uint16_t homing_debounce_delay;
@@ -130,13 +130,13 @@
 uint8_t settings_read_coord_data(uint8_t coord_select, float *coord_data);
 
 // Returns the step pin mask according to Grbl's internal axis numbering
-uint8_t get_step_pin_mask(uint8_t i);
+uint16_t get_step_pin_mask(uint8_t i);
 
 // Returns the direction pin mask according to Grbl's internal axis numbering
-uint8_t get_direction_pin_mask(uint8_t i);
+uint16_t get_direction_pin_mask(uint8_t i);
 
 // Returns the limit pin mask according to Grbl's internal axis numbering
-uint8_t get_limit_pin_mask(uint8_t i);
+uint16_t get_limit_pin_mask(uint8_t i);
 
 
 #endif
--- grbl-master/settings.c	2015-07-31 21:40:43.699147955 -0700
+++ grbl-master/settings.c	2015-07-31 21:42:12.168534345 -0700
@@ -297,7 +297,7 @@
 
 
 // Returns step pin mask according to Grbl internal axis indexing.
-uint8_t get_step_pin_mask(uint8_t axis_idx)
+uint16_t get_step_pin_mask(uint8_t axis_idx)
 {
   if ( axis_idx == X_AXIS ) { return((1<<X_STEP_BIT)); }
   if ( axis_idx == Y_AXIS ) { return((1<<Y_STEP_BIT)); }
@@ -306,7 +306,7 @@
 
 
 // Returns direction pin mask according to Grbl internal axis indexing.
-uint8_t get_direction_pin_mask(uint8_t axis_idx)
+uint16_t get_direction_pin_mask(uint8_t axis_idx)
 {
   if ( axis_idx == X_AXIS ) { return((1<<X_DIRECTION_BIT)); }
   if ( axis_idx == Y_AXIS ) { return((1<<Y_DIRECTION_BIT)); }
@@ -315,7 +315,7 @@
 
 
 // Returns limit pin mask according to Grbl internal axis indexing.
-uint8_t get_limit_pin_mask(uint8_t axis_idx)
+uint16_t get_limit_pin_mask(uint8_t axis_idx)
 {
   if ( axis_idx == X_AXIS ) { return((1<<X_LIMIT_BIT)); }
   if ( axis_idx == Y_AXIS ) { return((1<<Y_LIMIT_BIT)); }
