--- ./grbl/probe.c	2017-07-09 11:25:49.517234223 -0700
+++ ./grbl/probe.c	2017-07-09 11:27:24.321897254 -0700
@@ -28,13 +28,6 @@
 // Probe pin initialization routine.
 void probe_init()
 {
-  PROBE_DDR &= ~(PROBE_MASK); // Configure as input pins
-  #ifdef DISABLE_PROBE_PIN_PULL_UP
-    PROBE_PORT &= ~(PROBE_MASK); // Normal low operation. Requires external pull-down.
-  #else
-    PROBE_PORT |= PROBE_MASK;    // Enable internal pull-up resistors. Normal high operation.
-  #endif
-  probe_configure_invert_mask(false); // Initialize invert mask.
 }
 
 
@@ -50,8 +43,7 @@
 
 
 // Returns the probe pin state. Triggered = true. Called by gcode parser and probe state monitor.
-uint8_t probe_get_state() { return((PROBE_PIN & PROBE_MASK) ^ probe_invert_mask); }
-
+uint32_t probe_get_state() { return (debounce_rd() & PROBE_MASK) ^ probe_invert_mask; }
 
 // Monitors probe pin state and records the system position when detected. Called by the
 // stepper ISR per ISR tick.
--- ./grbl/probe.h	2017-07-09 11:27:37.173987136 -0700
+++ ./grbl/probe.h	2017-07-09 11:27:52.102091537 -0700
@@ -34,7 +34,7 @@
 void probe_configure_invert_mask(uint8_t is_probe_away);
 
 // Returns probe pin state. Triggered = true. Called by gcode parser and probe state monitor.
-uint8_t probe_get_state();
+uint32_t probe_get_state();
 
 // Monitors probe pin state and records the system position when detected. Called by the
 // stepper ISR per ISR tick.
