--- grbl-master/serial.h	2014-03-14 07:50:36.000000000 -0700
+++ grbl-master/serial.h	2014-06-06 10:07:48.000000000 -0700
@@ -20,33 +20,15 @@
 */
 
 /* This code was initially inspired by the wiring_serial module by David A. Mellis which
-   used to be a part of the Arduino project. */ 
+   used to be a part of the Arduino project. */
 
 #ifndef serial_h
 #define serial_h
 
 #include "nuts_bolts.h"
 
-#ifndef RX_BUFFER_SIZE
-  #define RX_BUFFER_SIZE 128
-#endif
-#ifndef TX_BUFFER_SIZE
-  #define TX_BUFFER_SIZE 64
-#endif
-
 #define SERIAL_NO_DATA 0xff
 
-#ifdef ENABLE_XONXOFF
-  #define RX_BUFFER_FULL 96 // XOFF high watermark
-  #define RX_BUFFER_LOW 64 // XON low watermark
-  #define SEND_XOFF 1
-  #define SEND_XON 2
-  #define XOFF_SENT 3
-  #define XON_SENT 4
-  #define XOFF_CHAR 0x13
-  #define XON_CHAR 0x11
-#endif
-
 void serial_init();
 
 void serial_write(uint8_t data);
@@ -56,4 +38,6 @@
 // Reset and empty data in read buffer. Used by e-stop and reset.
 void serial_reset_read_buffer();
 
+int serial_rx_hook(uint8_t data);
+
 #endif
