Specifying the KRMx01 machine and STM32F4-Discovery controller.
--- grbl-master/config.h	2014-03-14 07:50:36.000000000 -0700
+++ grbl-master/config.h	2014-05-25 10:19:50.000000000 -0700
@@ -29,14 +29,14 @@
 #define config_h
 
 // Default settings. Used when resetting EEPROM. Change to desired name in defaults.h
-#define DEFAULTS_GENERIC
+#define DEFAULTS_KRMX01
 
 // Serial baud rate
 #define BAUD_RATE 9600
 
 // Default pin mappings. Grbl officially supports the Arduino Uno only. Other processor types
 // may exist from user-supplied templates or directly user-defined in pin_map.h
-#define PIN_MAP_ARDUINO_UNO
+#define PIN_MAP_STM32F4_DISCOVERY
 
 // Define runtime command special characters. These characters are 'picked-off' directly from the
 // serial read data stream and are not passed to the grbl line execution parser. Select characters
