--- grbl-master/system.c	2015-07-17 07:19:57.000000000 -0700
+++ grbl-master/system.c	2015-08-01 07:45:08.404865939 -0700
@@ -23,14 +23,6 @@
 
 void system_init() 
 {
-  CONTROL_DDR &= ~(CONTROL_MASK); // Configure as input pins
-  #ifdef DISABLE_CONTROL_PIN_PULL_UP
-    CONTROL_PORT &= ~(CONTROL_MASK); // Normal low operation. Requires external pull-down.
-  #else
-    CONTROL_PORT |= CONTROL_MASK;   // Enable internal pull-up resistors. Normal high operation.
-  #endif
-  CONTROL_PCMSK |= CONTROL_MASK;  // Enable specific pins of the Pin Change Interrupt
-  PCICR |= (1 << CONTROL_INT);   // Enable Pin Change Interrupt
 }
 
 
@@ -38,9 +30,9 @@
 // only the realtime command execute variable to have the main program execute these when 
 // its ready. This works exactly like the character-based realtime commands when picked off
 // directly from the incoming serial data stream.
-ISR(CONTROL_INT_vect) 
+void control_isr(void) 
 {
-  uint8_t pin = (CONTROL_PIN & CONTROL_MASK);
+  uint32_t pin = debounce_rd() & CONTROL_MASK;
   #ifndef INVERT_CONTROL_PIN
     pin ^= CONTROL_MASK;
   #endif
