--- grbl-master/config.h	2014-03-30 07:33:46.000000000 -0700
+++ grbl-master/config.h	2014-03-30 07:34:47.000000000 -0700
@@ -36,7 +36,7 @@
 
 // Default pin mappings. Grbl officially supports the Arduino Uno only. Other processor types
 // may exist from user-supplied templates or directly user-defined in pin_map.h
-#define PIN_MAP_ARDUINO_UNO
+#define PIN_MAP_STM32F4_DISCOVERY
 
 // Define runtime command special characters. These characters are 'picked-off' directly from the
 // serial read data stream and are not passed to the grbl line execution parser. Select characters
